src/app/outbound.rs

```diff
@@ -77,6 +78,10 @@ impl tap::Inspect for Endpoint {
         self.metadata.tls_status()
     }
 
+    fn route_labels<B>(&self, req: &http::Request<B>) -> Option<Arc<IndexMap<String, String>>> {
+        req.extensions().get::<super::dst::Route>().map(|r| r.labels().clone())
```

Comment:
Does this need to clone the `Arc`? It looks like this could probably just return `&IndexMap`.

Reply:
to return &IndexMap, we'd need to change the signature to 
```
    fn route_labels<'r, B>(&self, req: &'r http::Request<B>) -> Option<&'r IndexMap<String, String>>
```
which feels like exposing an unfortunate implementation detail in the interface

Reply:
I think that's because the request stores the Arc'd Indexmap.. I couldn't get it to work, at least.